server/src/main/java/org/elasticsearch/index/IndexMode.java

```diff
@@ -257,6 +326,27 @@ protected static String tsdbMode() {
         }
     }
 
+    public static final CompressedXContent DEFAULT_LOGS_TIMESTAMP_MAPPING;
+
+    static {
+        try {
+            DEFAULT_LOGS_TIMESTAMP_MAPPING = new CompressedXContent(
+                ((builder, params) -> builder.startObject(MapperService.SINGLE_MAPPING_NAME)
+                    .startObject(DataStreamTimestampFieldMapper.NAME)
+                    .field("enabled", true)
+                    .endObject()
+                    .startObject("properties")
+                    .startObject(DataStreamTimestampFieldMapper.DEFAULT_PATH)
+                    .field("type", DateFieldMapper.CONTENT_TYPE)
+                    .endObject()
```

Comment:
Should `hostname` also be part of the default mapping?

Reply:
My understanding is that the `hostname` is required only if default index sorting applies. I expect that if an integration uses a different set of fields not including `hostname` (i.e. `(client_id, timestamp)`) then `hostname` is not required anymore. Integrations not using the `hostname` would break if we require it, right?

Reply:
Like we discussed, let's for now not define hostname in the builtin default mapping. If there is no hostname field mapping then logsdb will fail for now and this should trigger anyone to configure another index sort mode. Whether this is the behavior we would like to have as part of GA we can determine later.